filebeat/prospector/prospector.go

```diff
@@ -155,6 +158,13 @@ func (p *Prospector) createHarvester(state file.State) (*harvester.Harvester, er
 }
 
 func (p *Prospector) startHarvester(state file.State, offset int64) error {
+
+	if p.config.HarvesterLimit > 0 && atomic.LoadUint64(&p.harvesterCounter) >= p.config.HarvesterLimit {
+		return fmt.Errorf("Harvester limit reached.")
+	}
+
+	atomic.AddUint64(&p.harvesterCounter, 1)
```

Comment:
I think there is a race here if `startHarverster` can be called concurrently: two threads check the condition when the counter is equal to the limit-1, then they both increment it, so you get to limit+1. Not a big issue, but wouldn't things be simpler with a mutex?